integration-cli/docker_cli_run_test.go

```diff
@@ -78,6 +78,23 @@ func TestRunEchoStdoutWitCPULimit(t *testing.T) {
 }
 
 // "test" should be printed
+func TestRunEchoStdoutWitCPUQuota(t *testing.T) {
+	runCmd := exec.Command(dockerBinary, "run", "--cpu-quota", "8000", "busybox", "echo", "test")
```

Comment:
Can you docker inspect and verify whether this quota is actually put on the container? Even better would be to look under cgroup/cpu